crates/router/src/connector/paypal/transformers.rs

```diff
@@ -1213,19 +1283,59 @@ pub struct PaypalWebooksEventType {
     pub event_type: PaypalWebhookEventType,
 }
 
-impl From<PaypalWebhookEventType> for api::IncomingWebhookEvent {
-    fn from(event: PaypalWebhookEventType) -> Self {
-        match event {
-            PaypalWebhookEventType::PaymentCaptureCompleted
-            | PaypalWebhookEventType::CheckoutOrderCompleted => Self::PaymentIntentSuccess,
-            PaypalWebhookEventType::PaymentCapturePending
-            | PaypalWebhookEventType::CheckoutOrderProcessed => Self::PaymentIntentProcessing,
-            PaypalWebhookEventType::PaymentCaptureDeclined => Self::PaymentIntentFailure,
-            PaypalWebhookEventType::PaymentCaptureRefunded => Self::RefundSuccess,
-            PaypalWebhookEventType::PaymentAuthorizationCreated
-            | PaypalWebhookEventType::PaymentAuthorizationVoided
-            | PaypalWebhookEventType::CheckoutOrderApproved
-            | PaypalWebhookEventType::Unknown => Self::EventNotSupported,
+impl ForeignFrom<(PaypalWebhookEventType, Option<OutcomeCode>)> for api::IncomingWebhookEvent {
+    fn foreign_from((event, outcome): (PaypalWebhookEventType, Option<OutcomeCode>)) -> Self {
+        match (event, outcome) {
```

Comment:
Instead of `match (event, outcome)`, match on event alone and have a from implementation for outcome_code -> webhook_event_type